### ISSUE TITLE
Issue when starting with "Roads of the Rim (Continued)" or "Start Without Tech"

### ISSUE THREAD

**Issue by BubbleShy** [jptrrs/HumanResources#175]
Since the new update, whenever I start a game with either of these mods installed, it bugs out and deletes a majority or all of my starting pawns, animals, and equipment.

I don't have any game files as this was just a common issue I saw occuring while I was trying to figure out which mods were conflicting after the recent update. I could easily start some new games with just vanilla, your mod (and pre-reqs), and these mods to produce some files if you'd like.

I figured "Start Without Tech" would be a lost cause due to both being mods that mess with the research code, but it was working before the update.

**Comment by jptrrs** [comment]
Can you send me the logs for those failed attempts? And links to those mods, please.

**Comment by BubbleShy** [comment]
Here are the files for "Start Without Tech" in the order you asked:
https://gist.github.com/1bf4d941c084f7e3b87cfe7173302f2c
https://steamcommunity.com/sharedfiles/filedetails/?id=2550777561

In this one, the research tab is bugged. I can't select anything for them to research, even if I spawn in the research tables with dev mode.

And these are the files for "Roads of the Rim (Continued)":
https://gist.github.com/020f093f551d6bbc679bcf6c999408c3
https://steamcommunity.com/sharedfiles/filedetails/?id=2280318231

Whenever I start up a game with this mod, it will refuse to spawn in the starting characters, animals, and materials.


While I was creating these files, I also created a game without ResearchPal-Forked and Semi-Random Research. With both of the offending mods listed above on, I was able to get into the game and select a research objective fine, but the pawns refused to do any actual research. Also, no debug list ever populated whenever I combined these mods. I'm not sure how to force one to come up myself to check if there are any internal issues.

Let me know if you need anything else from my end, like that these weren't the files you were looking for.

**Comment by jptrrs** [comment]
Yes, those are what I was looking for. Thx.

With "Start Without Tech", I found no issues. Here's the the research selection pop-up, on the Semi-Random Research screen (since it's now compatible):
![20220622181405_1](https://user-images.githubusercontent.com/47060670/175139750-0fb7c3ba-fbb4-412a-ac2a-0c126c5c2643.jpg)
And here's the pawn actually researching:
![20220622181454_1](https://user-images.githubusercontent.com/47060670/175140096-99d17eda-f22c-4c47-9b3c-69ebe3729114.jpg)
Could it be you were looking in the wrong place?

As for "Roads of the Rim", I can confirm the issue. It happens at startup, it will show up as soon as the game initializes. Every issue you're having when starting a game is just a consequence of a failed initialization. This one was surprising to me, since I actually tested the update with RoR before releasing it! It seems I had something shielding me from the error.
I'm going to have to investigate it further.

**Comment by BubbleShy** [comment]
I'm not sure if I have something wrong with the mod settings, but this is what I get when I try to make a new game with "Start Without Tech":
![20220622192622_1](https://user-images.githubusercontent.com/107643069/175175206-e45d35a1-a9f4-4c19-b24f-ea9efee4293d.jpg)
I can open the actual research tree using the button at the top, but it still won't let me select a research project. Even when I disable "Prohibit normal project selection" in the "Semi Random Research" mod settings. At most, I can highlight the projects by right-clicking them, but left-clicking does nothing even with a simple research bench on the map and all of my pawns set to research in the Work tab.

I would like to share the file on this one, but a whole new problem has arisen on this particular file:
![20220622193730_1](https://user-images.githubusercontent.com/107643069/175177641-1af7ffda-7ca5-4711-9201-3c42534b892a.jpg)
I have pressed retry about a dozen times and nothing further happens.

Also, incase you couldn't tell by the other two screenshots:
![20220622192632_1](https://user-images.githubusercontent.com/107643069/175177748-2a0d5202-6869-4968-9bf4-70b5f370e40a.jpg)
This is another tribal start, and I only have 2 colonists (1 starting at the corner of the map), 1 animal, 200 silver, and a jade knife.


The only mods I have running (in order) are: Harmony, HugsLib, Human Resources, ResearchPal - Forked, Semi Random Research, and Start Without Tech.
I have Core, Royalty, and Ideology between Harmony and HugsLib.

**Comment by jptrrs** [comment]
This is weird. For those screenshots I shared earlier I followed your exact mod list. And your log also doesn't contain any errors.
Are you sure this is only when Human Resources is present? It could be something between Start Without Tech and Semi-Random Research. You can also try with each of those separately along HR.
Also, are you using the default tribal scenario?
Anyway, I should warn you: If SRR is present, no matter the settings, you'll never get the research option on the main research tree. Only studying and documenting.

**Comment by BubbleShy** [comment]
Thank you so much for helping me with this! I did some further testing and found that you were right about SWT doing this with other mods. Though, it wasn't a mod conflict. After doing a bit more digging, I realized that my dumbass set SWT to consider my colony at an "Animal" level of tech. Because of this, the mod was blocking any and all chances for me to actually do any research.

I apologize profusely for wasting your time on this! But thank you immensely for your replies! Without them I would be lost.

As for RotR, the workshop page for the mod itself states you can add it to existing saves. I tried just that with my current playthrough that included an extensive mod-list (with HR), and nothing seemed to break like these startup examples. Maybe this is why the error was shielded from your tests? Because if you didn't start a new file to test the update with RotR, it wouldn't have come up for you. (I'm not being accusatory; I genuinely see no reason why you wouldn't use an existing file in testing)

Now that I know what's wrong with SWT, I may just start a new colony with it and add RotR after everything is set up. If you plan to look deeper into the issue, I wish you all the luck and I hope you find what's bugging out. I would be lying if I said I wasn't curious what's going on here.

Though, I wouldn't blame you if you put a warning on your mod's Workshop page about starting with RotR and call it a day.

**Comment by jptrrs** [comment]
Don't worry, I'm glad you managed to sort it out.
Also, it was good you uncovered that issue with RotR. A similar error turned up on issue #176, but without the care for isolating the conflict. With your test, I now have enough evidence to pinpoint the actual issue. A fix is on its way.
What happened was this: I have this "Special Cases" routine to deal with bunch of outliers, including RotR. It only triggers when one of those mods is present. So, when I decided to include some stuff regarding "Vanilla Furniture Expanded - Art", that was where I placed it as well. The thing is, unlike the others, the code I wrote for VFE-Art could only work if it was present. So, if you had RotR but not VFE-Art, that routine would trigger, but it would freak out when it couldn't find the VFE stuff, causing a catastrophic error.  This would botch the startup process, rendering the mod unusable for any subsequent game.
It took me a while to figure it out, but it's an easy fix. I'll upload it to the workshop in a minute.
